cpp/src/sort/sort.cuh

```diff
@@ -0,0 +1,70 @@
+// -*-c++-*-
+
+#pragma once
+
+#include <cudf/types.h>
+
+namespace cusort {
+
+  /**
+   * @brief  Sort key value pairs distributed across multiple GPUs
+   *
+   *   This sort function takes arrays of keys and values distributed
+   *   around multiple GPUs 
+   *
+   * @param[in]   input_keys_d     The unsorted keys, stored in device arrays.
+   *                               input_keys_d[i] is the array of keys on GPU i
+   * @param[in]   input_values_d   The unsorted values, stored in device arrays.
+   *                               input_values_d[i] is the array of values on GPU i
+   * @param[in]   input_length_h   Host array containing the number of elements on
+   *                               each GPU in the input key/value arrays.
+   * @param[out]  output_keys_d    The sorted keys, stored in device arrays.
+   *                               output_keys_d[i] is the array of keys on GPU i
+   * @param[out]  output_values_d  The sorted values, stored in device arrays.
```

Comment:
I suppose the output is allocated by this function. I would recommend adding a note mentioning that.
Also, I recall that one key cannot be present on two GPU in the output. This is a feature that worth documenting I think.

Reply:
Will add to documentation tomorrow.